src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js

```diff
@@ -5,39 +5,71 @@ import 'whatwg-fetch';
 
 import './style.scss';
 
-function handleDownload(url, filename, accessToken) {
-  const headers = {};
+class DownloadButton extends React.Component {
+  constructor(props) {
+    super(props);
 
-  if (accessToken) {
-    headers.Authorization = `Bearer ${accessToken}`;
+    this.downloadLink = React.createRef();
+    this.handleDownload = this.handleDownload.bind(this);
   }
 
-  fetch(url, {
-    method: 'GET',
-    headers,
-    responseType: 'blob'
-  }).then((response) => response.blob())
-    .then((blob) => {
-      const link = document.querySelector('.download-button__link');
-      link.href = window.URL.createObjectURL(blob);
-      link.click();
-    });
-}
+  handleDownload(url, filename, accessToken) {
+    const headers = {};
+
+    if (accessToken) {
+      headers.Authorization = `Bearer ${accessToken}`;
+    }
+
+    fetch(url, {
+      method: 'GET',
+      headers,
+      responseType: 'blob'
+    }).then((response) => response.blob())
+        .then((blob) => {
+          if (navigator.msSaveOrOpenBlob) {
+            navigator.msSaveOrOpenBlob(blob, filename);
+          } else {
+            const link = this.downloadLink.current;
+            link.href = window.URL.createObjectURL(blob);
+            link.click();
+          }
+        });
+  }
 
-const DownloadButton = ({ label, url, filename, accessToken }) => (
-  <div className="download-button align-self-center">
-    <button
-      className="incident-detail__button"
-      onClick={() => handleDownload(url, filename, accessToken)}
-    >{label}</button>
-
-    <a
-      href="#"
-      className="download-button__link"
-      download={filename}
-    ></a>
-  </div>
-);
+  render() {
+    const { label, url, filename, accessToken } = this.props;
+    return (
+      <div className="download-button align-self-center">
+        <button
+          className="incident-detail__button"
+          onClick={() => this.handleDownload(url, filename, accessToken)}
+        >{label}</button>
+
+        <a
```

Comment:
Why both a `<button>` and an `<a>` element? Should be safe to remove the `<button>` and handle the click on the `<a>`.

Reply:
alleen de anchor werkte niet, maar alleen de button wel: deze test op Chrome, Safari, FF en op Win IE11, en edge 15 en FF